### PR TITLE
feat(eval): add generalized zero-shot object detection support

### PR DESCRIPTION
**Pull request by ssss141414**
## Summary

This PR adds generalized `zero-shot-object-detection` eval support for `google/owlv2-base-patch16-finetuned` through eval command/config wiring plus tests.

Final candidate and PR head are now `bfe94515d6c85151486a2eb8929e8722e62b0991`.

Scope/outcome:
- Producer scope is CLI/config wiring for query budget (`--max-queries`) and related tests.
- Reused immutable provenance from `a723405f8dc8864c0c0a0aa754a41bddeb2c8a56` for `L0/L1/L2/analyze` artifacts.
- Reran `L3` smoke and core quality gates on final candidate `bfe94515`.
- Highest goal verdict: `L3 PASS`.

### What the model does
- OWLv2 is an open-vocabulary detector: image + text queries in, query-conditioned detection outputs out.

### Supported tasks
- `zero-shot-object-detection` (checkpoint/transformers/optimum-onnx/winml surfaces).
- `feature-extraction` (optimum-onnx/winml surfaces).

### Architecture (encoding-safe tree)
```text
Owlv2ForObjectDetection
+- Vision encoder backbone (patch embeddings + transformer blocks)
+- Text encoder backbone (token embeddings + transformer blocks)
+- Cross-modal alignment projections (text/image embeddings)
`- Detection heads
   +- logits [image_batch, num_queries, 1]
   `- pred_boxes [image_batch, num_queries, 4]
```

### Baseline and goal
- Pinned main baseline: `2b9ec0e9e57bba8003d25eaff85f8c7c9e30d75a` (WinML `0.3.0`).
- Committed effort/goal: Effort `L2`, goal ceiling `L3`.
- Final outcome: shipped `L3`, highest goal verdict `L3 PASS`.

### Goal ladder
- `L0 PASS` (reused from `a723405f`): both required CPU tuples built and structurally validated.
- `L1 PASS` (reused from `a723405f`): perf completed for both tuples.
- `L2 PASS` (reused from `a723405f`): PT-vs-ONNX parity completed on real decoded COCO image.
- `L3 PASS` (rerun on `bfe94515`): public CLI smoke with bounded query budget.

| Tier | EP / Device | Precision | Verdict | Mean (ms) | p50 (ms) | Throughput (samples/s) | ΔRAM (MB) | ΔVRAM (MB) |
|---|---|---|---|---:|---:|---:|---:|---|
| L0/L1 | CPUExecutionProvider / cpu | fp32 | PASS | 3253.185 | 3182.379 | 0.31 | 908.88 | null |
| L0/L1 | CPUExecutionProvider / cpu | fp16 | PASS | 5527.614 | 5770.852 | 0.18 | 3027.47 | null |

FP16 is slower than FP32 in this run.

### Final-SHA public smoke (exact)
Command (public-style, no absolute paths):
```powershell
$OUT='temp/pr1217-bfe94515-repro'
New-Item -ItemType Directory -Path $OUT -Force | Out-Null
uv run winml eval -m temp/pr1217-rerun/tester/a723405f-r1/artifacts/fp32/model.onnx --model-id google/owlv2-base-patch16-finetuned --task zero-shot-object-detection --ep cpu --device cpu --samples 1 --dataset detection-datasets/coco --dataset-revision cf0b22332314a937e9dc8a1957b21725430bb41d --split val --no-shuffle --max-queries 3 --trust-remote-code --output $OUT/l3_smoke_eval_maxq3_cli_public.json --overwrite
```
Observed accounting (exact):
- `requested=1`, `selected=1`, `processed=1`, `decoded=1`
- `query_requested=3`, `query_available=80`, `query_used=3`, `query_truncated=77`, `query_passes=3`
- Metric: `map=0.0` (functional smoke semantics only)

### L2 parity breakdown (reused provenance from `a723405f`)
Tester contract note: there is no hard L2 threshold; values are reported as observed.

FP32:
- `logits`: cosine `0.9999999999548849`, max_abs `0.0019044876098632812`
- `pred_boxes`: cosine `0.9999999999735265`, max_abs `0.00014513731002807617`
- `text_embeds`: cosine `0.999999999999833`, max_abs `1.1920928955078125e-07`
- `image_embeds`: cosine `0.9999999999754301`, max_abs `0.0009830594062805176`
- aggregate: min_cosine `0.9999999999548849`, max_abs `0.0019044876098632812` (from `logits`)

FP16:
- `logits`: cosine `0.9999742966056367`, max_abs `1.5570344924926758`
- `pred_boxes`: cosine `0.9999867989005999`, max_abs `0.08713313937187195`
- `text_embeds`: cosine `0.9999998139610052`, max_abs `8.073821663856506e-05`
- `image_embeds`: cosine `0.9999897980460138`, max_abs `0.6433656215667725`
- aggregate: min_cosine `0.9999742966056367`, max_abs `1.5570344924926758` (from `logits`)

Explicit requested detail:
- logits produced `fp32 max_abs = 0.0019044876098632812`
- logits produced `fp16 max_abs = 1.5570344924926758`
- cosines are recorded above exactly as observed.

## Bug fix explanation

User-visible issue:
- Query budget control was misleading in prior path (`--column max_queries=...`) and could fan out unexpectedly.

Fix details:
- Dedicated CLI option added: `--max-queries`.
- Wiring path: command option -> config override -> `DatasetConfig.max_queries` -> evaluator query budget.
- Default bounded behavior is explicit: default is `16`.
- Explicit full-vocabulary mode remains available via config: `max_queries: null`.
- `--column` remains column mapping only (no query-budget semantic overload).
- Evaluator keeps annotation/category filtering semantics for authoritative labels and query-capacity handling.
- Regression tests added/updated for CLI mapping, invalid values, schema/help text, and evaluator delivery path.

Changed files (producer scope):
- `src/winml/modelkit/commands/eval.py`
- `src/winml/modelkit/utils/eval_utils.py`
- `tests/unit/eval/test_eval.py`
- `tests/unit/commands/test_eval.py`

## Analyze and quality gates

- Analyze status: `ANALYZE-PARTIAL-SUCCESS` with non-zero exits but complete JSON payloads consumed.
- Reused analyze provenance: source candidate `a723405f`.
- Rerun quality gates on final candidate `bfe94515`:
  - `ruff`: pass
  - `mypy`: pass
  - query-budget focused tests: pass
  - eval CLI/command focused tests: pass
  - commands partition: pass
  - integration ZSOD pinned-remote: pass
  - models/pipeline partitions: pass (reused provenance from `a723405f`)

## Public reproduce block

```powershell
$OUT='temp/pr1217-bfe94515-repro'
New-Item -ItemType Directory -Path $OUT -Force | Out-Null
uv run winml eval -m temp/pr1217-rerun/tester/a723405f-r1/artifacts/fp32/model.onnx --model-id google/owlv2-base-patch16-finetuned --task zero-shot-object-detection --ep cpu --device cpu --samples 1 --dataset detection-datasets/coco --dataset-revision cf0b22332314a937e9dc8a1957b21725430bb41d --split val --no-shuffle --max-queries 3 --trust-remote-code --output $OUT/l3_smoke_eval_maxq3_cli_public.json --overwrite
uv run pytest tests/unit/eval/test_zero_shot_object_detection_evaluator.py -k 'query_budget or max_queries or out_of_cap or schema_documents_authoritative_categories_and_prompt or default_dataset_is_pinned_coco_xywh' --tb=short --no-cov
uv run pytest tests/unit/eval/test_eval.py -k 'max_queries or TestEvalCli' --tb=short --no-cov
uv run pytest tests/unit/commands/test_eval.py -k 'max_queries or schema_shows_max_queries_without_column_override or max_queries_reaches_evaluator or TestEvalHelp' --tb=short --no-cov
uv run pytest tests/unit/commands tests/unit/config tests/unit/build tests/unit/compiler tests/unit/session tests/unit/eval --tb=short --no-cov -m 'not e2e and not npu and not gpu'
uv run pytest tests/integration/eval/test_zero_shot_object_detection.py --tb=short --no-cov
uv run ruff check src/ tests/
uv run mypy -p winml.modelkit
```

## Owner communication record (internal reviewer findings closure)

No reviewer comment was posted; closure is recorded in canonical body and final evidence.

- Processor trust present at exact code path: public smoke command includes `--trust-remote-code` and uses final candidate pathing.
- Reproduce block present: public commands are included above and use `--max-queries 3` without absolute paths.
- Query budget fixed: final candidate verifies exact accounting `requested=3 available=80 used=3 truncated=77 passes=3`.
- Parity explained: per-output fp32/fp16 cosine and max_abs values are documented, including logits-specific values.
- Live gates rerun expectation: post-push CI gates for `bfe94515` are rerunning now (pending in live snapshot below).

## PR state snapshot

- Draft state: `true` (kept as Draft)
- Label retained: `model-scale-by-skill`
- Head branch: `ssss141414/add-zero-shot-object-detection-eval`
- Head SHA: `bfe94515d6c85151486a2eb8929e8722e62b0991`
- Open review threads: `0`
- Issue comments: `0`
- Reviews: `0`

Checks snapshot at update time:
- license/cla: pass
- Analyze (Python): pending
- lint: pending
- test (analyze): pending
- test (commands): pending
- test (models): pending
- test (optim): pending
- test (remaining): pending

### CODE REVIEW

**Comment by ssss141414** [comment]
APPROVE (independent reviewer)

Reviewed exact PR head SHA: bfe94515d6c85151486a2eb8929e8722e62b0991

Independent checks completed on this SHA:
- Diff scope reviewed line-by-line against origin/main...HEAD.
- Focused evaluator/CLI tests passed (201 passed), including zero-shot object detection integration.
- Touched-file Ruff passed.
- Canonical mypy passed (uv run mypy -p winml.modelkit).
- Live GitHub checks for this exact SHA are all COMPLETED/SUCCESS.
- Review threads enumerated: 0 open (0 total). Conversation and line comments: 0.

No blocking code or evidence findings in this review pass.
